src/ddmd/parse.d

```diff
@@ -4795,6 +4798,202 @@ final class Parser(AST) : Lexer
            ddmd.errors.deprecation(loc, "instead of C-style syntax, use D-style syntax '%s%s%s'", t.toChars(), sp, s);
     }
 
+    private static template ParseForeachArgs(bool isStatic, bool isDecl)
+    {
+        static alias Seq(T...) = T;
+        static if(isDecl)
+        {
+            alias ParseForeachArgs = Seq!(AST.Dsymbol*);
+        }
+        else
+        {
+            alias ParseForeachArgs = Seq!();
+        }
+    }
+    static template ParseForeachRet(bool isStatic, bool isDecl)
```

Comment:
It is static because it does not need to be nested inside the enclosing `struct`. I can also remove `static` if that is better (it should not make a difference here).